Lib/importlib/_bootstrap.py

```diff
@@ -263,7 +263,13 @@ def _requires_builtin_wrapper(self, fullname):
 def _requires_frozen(fxn):
     """Decorator to verify the named module is frozen."""
     def _requires_frozen_wrapper(self, fullname):
-        if not _imp.is_frozen(fullname):
+        if not _imp.is_frozen(fullname) or (
+            # bpo-45272: os.path is not a concrete module, but an importable
+            #            attribute of 'os'. _imp.is_frozen('os.path') returns
+            #            False but it is frozen, so we check the loader here.
+            fullname in sys.modules
+            and isinstance(sys.modules[fullname].__spec__.loader, FrozenImporter)
```

Comment:
@ericsnowcurrently, couldn't we do `loader.origin == 'frozen'` instead? Are we planing to change the value of origin to the actual file or something like that?